Added build on launch

maeday-modus opened the pull request:
I'm not an expert on how everyone else organizes their projects and workspaces, so I'm not 100% positive this will work for every code base. Here's a summary of the changes:
1. `launch.json` and `tasks.json` contents now generate into `.code-workspace` file.
    * Lets you run the build command from the workspace root directory, independent of where project folders are _(sub-directories, sub-sub-directories, etc.)_
    * Launch configs generate based on the `startproject` specified by the user in their premake script.
1. Moved launch and task generation code into their own files to let vscode_workspaces access both.
1. Launch configs generate with a `"preLaunchTask":` that calls respective build task.
     * Might want to also generate launch configs that don't build in case user wants to run without re-building.

maeday-modus committed:
Made launch work with build tasks

maeday-modus committed:
Removed extra path in .code-workspace file

maeday-modus committed:
Hopefully fixed multi-project build tasks not being recognized

maeday-modus committed:
Moved task generation to projects - kind of a dumb hack

maeday-modus committed:
Moved launch and tasks to generate in the .code-workspace file

maeday-modus committed:
Updated manifest

maeday-modus committed:
Only generate main workspace folder

maeday-modus committed:
Added check for startproject

maeday-modus committed:
Made sure launch options are generated even if startproject isn't specified

peter1745 commented:
Hey! I don't really actively maintain this project anymore due to me moving away from VS Code, however if you're finding this useful and is willing to maintain it I'd be happy to make your fork the main repo if that's alright with you 🙂

maeday-modus commented:
Yeah man! I'd be happy to if that works better for you:)

peter1745 commented:
> Yeah man! I'd be happy to if that works better for you:)

Awesome, I'll archive this repo and update the readme to link to your fork of it 🙂

maeday-modus committed:
Update README.md

Added reference to original project + general updates

maeday-modus committed:
Added Comments and Updated README.md

Comments help me understand the existing code better. Also better for future maintenance.
